pkg/secrets/fetch_secret.go

```diff
@@ -49,6 +54,41 @@ func execCommand(inputPayload string) ([]byte, error) {
 	defer cancel()
 
 	cmd := exec.CommandContext(ctx, secretBackendCommand, secretBackendArguments...)
+
+	if secretBackendCommandSHA256 != "" {
+		if !filepath.IsAbs(secretBackendCommand) {
+			return nil, fmt.Errorf("error while running '%s': absolute path required with SHA256", secretBackendCommand)
+		}
```

Comment:
Please consider checking if the target file is a symlink too. The agent must not be able to run a target binary pointing to another one via a symlink. Symbolic link attacks can be used to change a file's permissions, corrupt data, destroy data, or as a DoS attack.

Reply:
Great point, will add this check, this however will only take place when `secret_backend_command_sha256` is provided.

Reply:
Added this check.